### PR TITLE
Add show method for FieldTuple

### LINKED ISSUE #1256: Pretty printing for named tuples of fields
Named tuples of fields are a very common thing to look at in the REPL but currently it's hard to parse visually and contains a lot of noise, e.g.

```julia
julia> fields(model)
(uh = Field located at (Face, Cell, Cell)
├── data: OffsetArrays.OffsetArray{Float64,3,Array{Float64,3}}, size: (7, 7, 7)
├── grid: RegularCartesianGrid{Float64, Periodic, Periodic, Bounded}(Nx=1, Ny=1, Nz=1)
└── boundary conditions: x=(west=Periodic, east=Periodic), y=(south=Periodic, north=Periodic), z=(bottom=ZeroFlux, top=ZeroFlux), vh = Field located at (Cell, Face, Cell)
├── data: OffsetArrays.OffsetArray{Float64,3,Array{Float64,3}}, size: (7, 7, 7)
├── grid: RegularCartesianGrid{Float64, Periodic, Periodic, Bounded}(Nx=1, Ny=1, Nz=1)
└── boundary conditions: x=(west=Periodic, east=Periodic), y=(south=Periodic, north=Periodic), z=(bottom=ZeroFlux, top=ZeroFlux), h = Field located at (Cell, Cell, Cell)
├── data: OffsetArrays.OffsetArray{Float64,3,Array{Float64,3}}, size: (7, 7, 7)
├── grid: RegularCartesianGrid{Float64, Periodic, Periodic, Bounded}(Nx=1, Ny=1, Nz=1)
└── boundary conditions: x=(west=Periodic, east=Periodic), y=(south=Periodic, north=Periodic), z=(bottom=ZeroFlux, top=ZeroFlux))
```

Something better could be

```julia
julia> fields(model)
NamedTuple with 3 fields: uh, vh, h
```

or for a bit more information

```julia
NamedTuple with 3 fields:
├── uh: Field located at (Face, Cell, Cell)
├── vh: Field located at (Cell, Face, Cell)
└── h: Field located at (Cell, Cell, Cell)
```

### PR DESCRIPTION
**Pull request by glwagner**
So that

```julia
julia> using Oceananigans

julia> grid = RectilinearGrid(size=(4, 4, 4), extent=(4, 4, 4));

julia> a = CenterField(grid); b = XFaceField(grid);

julia> (; a, b)
(a = 4×4×4 Field{Center, Center, Center} on RectilinearGrid on CPU
├── grid: 4×4×4 RectilinearGrid{Float64, Periodic, Periodic, Bounded} on CPU with 1×1×1 halo
├── boundary conditions: west=Periodic, east=Periodic, south=Periodic, north=Periodic, bottom=ZeroFlux, top=ZeroFlux, immersed=ZeroFlux
└── data: 6×6×6 OffsetArray(::Array{Float64, 3}, 0:5, 0:5, 0:5) with eltype Float64 with indices 0:5×0:5×0:5
    └── max=0.0, min=0.0, mean=0.0, b = 4×4×4 Field{Face, Center, Center} on RectilinearGrid on CPU
├── grid: 4×4×4 RectilinearGrid{Float64, Periodic, Periodic, Bounded} on CPU with 1×1×1 halo
├── boundary conditions: west=Periodic, east=Periodic, south=Periodic, north=Periodic, bottom=ZeroFlux, top=ZeroFlux, immersed=ZeroFlux
└── data: 6×6×6 OffsetArray(::Array{Float64, 3}, 0:5, 0:5, 0:5) with eltype Float64 with indices 0:5×0:5×0:5
    └── max=0.0, min=0.0, mean=0.0)
```

becomes

```julia
julia> (; a, b)
NamedTuple with 2 Fields on 4×4×4 RectilinearGrid{Float64, Periodic, Periodic, Bounded} on CPU with 1×1×1 halo:
├── a: 4×4×4 Field{Center, Center, Center} on RectilinearGrid on CPU
└── b: 4×4×4 Field{Face, Center, Center} on RectilinearGrid on CPU
```

If the fields are on different grids (an edge case) then:

```julia
julia> c = CenterField(RectilinearGrid(size=(5, 5, 5), extent=(1, 1, 1)));

julia> (; a, b, c)
NamedTuple with 3 Fields on different grids:
├── a: 4×4×4 Field{Center, Center, Center} on RectilinearGrid on CPU
│   └── grid: 4×4×4 RectilinearGrid{Float64, Periodic, Periodic, Bounded} on CPU with 1×1×1 halo
├── b: 4×4×4 Field{Face, Center, Center} on RectilinearGrid on CPU
│   └── grid: 4×4×4 RectilinearGrid{Float64, Periodic, Periodic, Bounded} on CPU with 1×1×1 halo
└── c: 5×5×5 Field{Center, Center, Center} on RectilinearGrid on CPU
    └── grid: 5×5×5 RectilinearGrid{Float64, Periodic, Periodic, Bounded} on CPU with 1×1×1 halo
```

Closes #1256

### CODE REVIEW

**Review by navidcy — APPROVED** [review]
@glwagner you are on fire with `show()`

**Comment by glwagner** [comment]
Shoulda known this was playing with fire... 😄  lucky for #2401 .
